server.go

```diff
@@ -1680,6 +1683,12 @@ func newServer(cfg *Config, listenAddrs []net.Addr,
 	return s, nil
 }
 
+// UpdateEstimatorValue is a callback function to change estimator value in the  
+// main cfg when the lncli setmccfg is called to change the estimator value. 
+ func (s *server) UpdateEstimatorValue(estimator routing.Estimator) {
+	s.cfg.Estimator = estimator
```

Comment:
You're just updating the estimator interface here, which is just used as a cache after parsing the config values into the correct structure.
Try something like this instead (untested, but you should get the idea):
```
	switch c := estimator.Config().(type) {
	case *routing.AprioriConfig:
		targetCfg := s.cfg.SubRPCServers.RouterRPC.AprioriConfig
		targetCfg.PenaltyHalfLife = c.PenaltyHalfLife
		targetCfg.Weight = c.AprioriWeight
		targetCfg.CapacityFraction = c.CapacityFraction
		targetCfg.HopProbability = c.AprioriHopProbability

	case *routing.BimodalConfig:
		targetCfg := s.cfg.SubRPCServers.RouterRPC.BimodalConfig
		targetCfg.Scale = int64(c.BimodalScaleMsat)
		targetCfg.NodeWeight = c.BimodalNodeWeight
		targetCfg.DecayTime = c.BimodalDecayTime
	}
```